src/server/aes_cbc_block/cbe-pool.adb

```diff
@@ -0,0 +1,263 @@
+--
+-- Copyright (C) 2019 Genode Labs GmbH, Componolit GmbH, secunet AG
+--
+-- This file is part of the Consistent Block Encrypter project, which is
+-- distributed under the terms of the GNU Affero General Public License
+-- version 3.
+--
+
+pragma Ada_2012;
+
+package body CBE.Pool
+with Spark_Mode
+is
+
+	package body Item
+	with Spark_Mode
+	is
+		--
+		-- Primitive_Completed
+		--
+		procedure Primitive_Completed(
+			Obj  : in out Item_Type;
+			Prim :        Primitive.Object_Type)
+		is
+			use Request;
+			Rq : Request.Object_Type;
+		begin
+			if not Primitive.Success(Prim) then
+				Rq := Item.Req(Obj);
+				Request.Success(Rq, False);
+				Item.Req(Obj, Rq);
+			end if;
+			Obj.Nr_Of_Done_Prims := Obj.Nr_Of_Done_Prims + 1;
```

Comment:
Is an overflow in this addition desirable? If not you should change `Number_Of_Primitives_Type` to a ranged type (and then check that no overflow can occur).